sitesearch/api/search.py

```diff
@@ -73,52 +83,46 @@ class SearchResource(Resource):
                   If this isn't specified, the query searches the default site specified in
                   AppConfiguration. E.g. https://example.com/search?q=python&site_url=https://docs.redislabs.com
     """
-    def on_get(self, req, resp):
-        """Run a search."""
-        query = req.get_param('q', default='')
-        from_url = req.get_param('from_url', default='')
-        start = int(req.get_param('start', default=0))
-        site_url = req.get_param('site', default=None)
-        query_len = len(query)
-
-        if query_len == 2 and query[1] == '*':
-            char = query[0]
-            if char in SINGLE_CHAR_MAP:
-                query = f"{SINGLE_CHAR_MAP[query[0]]}*"
-
-        # Return an error if a site URL was given but it's invalid.
-        if site_url and site_url not in self.app_config.sites:
-            raise falcon.HTTPBadRequest(
-                "Invalid site", "You must specify a valid search site.")
-
-        # Use the default search site if no site URL was given.
-        if not site_url:
-            site_url = self.app_config.default_search_site.url
-
-        search_site = self.app_config.sites.get(site_url)
-        section = indexer.get_section(site_url, from_url)
-
-        try:
-            num = min(int(req.get_param('num', default=DEFAULT_NUM)), MAX_NUM)
-        except ValueError:
-            num = DEFAULT_NUM
-
-        index_alias = self.keys.index_alias(search_site.url)
-        search_client = get_search_connection(index_alias)
-        q = parse(query, section, search_site).paging(start, num)
-
-        start = time.time()
-        try:
-            res = search_client.search(q)
-        except (redis.exceptions.ResponseError, UnicodeDecodeError) as e:
-            log.error("Search query failed: %s", e)
-            total = 0
-            docs = []
-        else:
-            docs = res.docs
-            total = res.total
-        end = time.time()
-        newrelic.agent.record_custom_metric('search/query_ms', end - start)
-
-        docs = transform_documents(docs, search_site, q.query_string())
-        resp.body = json.dumps({"total": total, "results": docs})
+    from_url = from_url if from_url else ''
+    start = start if isinstance(start, int) else 0
```

Comment:
Is `isinstance` necessary here, since `start` has a type hint of `Optional[int]`? FastAPI uses Pydantic under the hood, which will attempt to coerce anything passed-in to the function to an `int`.

> pydantic uses int(v) to coerce types to an int; see this warning on loss of information during data conversion

[https://pydantic-docs.helpmanual.io/usage/types/](https://pydantic-docs.helpmanual.io/usage/types/)